service/worker/service.go

```diff
@@ -449,17 +456,21 @@ func (s *Service) startBatcher() {
 	}
 }
 
-func (s *Service) startScanner() {
-	sc := scanner.New(
+func (s *Service) initScanner() {
+	s.scanner = scanner.New(
```

Comment:
What is this change for?

Reply:
This allows us to close the scanner when the service closes (which waits for all of its coroutines to finish)